File: frontend/app/src/main/java/com/example/socialize/models/Message.kt
package com.example.socialize.models

data class Message(
    val id: Int = 0,
    val text: String,
    val author: String?,
    val sentAt: String?,
    val fromId: Int,
    val toId: Int = 0,
    val isGroupChatMessage: Boolean = true,
    val toGroupId: Int,
    val userIsAuthor: Boolean = true
)

File: backend/models/message.js
'use strict';
const {
    Model
} = require('sequelize');

module.exports = (sequelize, DataTypes) => {
    class Message extends Model {
        static associate(models) {
            Message.belongsTo(models.User, {
                as: 'from',
                foreignKey: 'fromId'
            });
    
            Message.belongsTo(models.User, {
                as: 'to',
                foreignKey: 'toId'
            });

            Message.belongsTo(models.Group, {
                as: 'toGroup',
                foreignKey: 'toGroupId'
            });
        }
    }

    Message.init({
        text: DataTypes.STRING,
        fromId: DataTypes.INTEGER,
        toId: DataTypes.INTEGER,
        isGroupChatMessage: DataTypes.BOOLEAN,
        toGroupId: DataTypes.INTEGER
    }, {
        sequelize,
        modelName: 'Message',
        tableName: 'message'
    });

    return Message;
};